test/web-auth/transaction-manager.test.js

```diff
@@ -83,59 +90,87 @@ context('TransactionManager', function() {
-      expect(Storage.prototype.setItem.lastCall.args[0]).to.be('com.auth0.auth.randomString');
-      expect(Storage.prototype.setItem.lastCall.args[1]).to.be.eql({
-        nonce: null,
-        appState: 'appState',
-        state: 'randomString',
-        lastUsedConnection: null
-      });
-    });
-    it('only stores random nonce when generateNonce is true', function() {
-      var result = this.tm.generateTransaction('appState', null, null, null, true);
-      expect(result).to.be.eql({ state: 'randomString', nonce: 'randomString' });
-      expect(Storage.prototype.setItem.calledOnce).to.be(true);
-      expect(Storage.prototype.setItem.lastCall.args[0]).to.be('com.auth0.auth.randomString');
-      expect(Storage.prototype.setItem.lastCall.args[1]).to.be.eql({
-        nonce: 'randomString',
-        appState: 'appState',
-        state: 'randomString',
-        lastUsedConnection: null
-      });
-    });
-    it('uses nonce/state when provided', function() {
-      var result = this.tm.generateTransaction('appState', 'state', 'nonce', null);
-      expect(result).to.be.eql({ state: 'state', nonce: 'nonce' });
-      expect(Storage.prototype.setItem.calledOnce).to.be(true);
-      expect(Storage.prototype.setItem.lastCall.args[0]).to.be('com.auth0.auth.state');
-      expect(Storage.prototype.setItem.lastCall.args[1]).to.be.eql({
-        nonce: 'nonce',
-        appState: 'appState',
-        state: 'state',
-        lastUsedConnection: null
-      });
-    });
-    it('uses lastUsedConnection when provided', function() {
-      var result = this.tm.generateTransaction('appState', 'state', 'nonce', 'lastUsedConnection');
-      expect(result).to.be.eql({ state: 'state', nonce: 'nonce' });
-      expect(Storage.prototype.setItem.calledOnce).to.be(true);
-      expect(Storage.prototype.setItem.lastCall.args[0]).to.be('com.auth0.auth.state');
-      expect(Storage.prototype.setItem.lastCall.args[1]).to.be.eql({
-        nonce: 'nonce',
-        appState: 'appState',
-        state: 'state',
-        lastUsedConnection: 'lastUsedConnection'
-      });
-    });
-    it('stores state with expires option equal to 30 mins', function() {
-      this.tm.generateTransaction('appState', 'state', 'nonce', null);
-      expect(Storage.prototype.setItem.calledOnce).to.be(true);
-      expect(typeof Storage.prototype.setItem.lastCall.args[2]).to.be('object');
-      expect(Storage.prototype.setItem.lastCall.args[2]).to.be.eql({ expires: times.MINUTES_30 });
+    context('when inside the hosted login page', function() {
+      beforeEach(function() {
+        windowHelper.getWindow.restore();
+        stub(windowHelper, 'getWindow', function() {
+          return { location: { host: 'auth.myapp.com' } };
+        });
+        this.tm = new TransactionManager({ domain: 'auth.myapp.com' });
+      });
+      it('uses nonce/state when provided', function() {
+        var result = this.tm.generateTransaction('appState', 'state', 'nonce', null);
```

Comment:
Not necessary to change but using the same value as the key is a little confusing to read and can easily lead to the tests not being written incorrectly. Suggest `providedState` and `providedNonce` or something like that.

Reply:
pushed this change